### ISSUE TITLE
Remote PageHandler clears selection

### ISSUE THREAD

**Issue by sgoth** [vincjo/datatables#75]
remote/handlers/PageHandler.ts calls `this.selected.set([])` in goto whereas the local version doesn't.

Is there a reason for that? I'd like to keep my selection model intact even through page or filter changes.

**Comment by vincjo** [comment]
`SelectHandler` would need to keep track of selected rows per each page instead of a global array, to know if a current page is in "allSelected" state.

So far: 
```ts
$selected = [row1, row2, row3 ...]
```

Required: 
```ts
$selected = {
    page1: [row1, row2, row3 ...],
    page2: [row1, row2, row3 ...],
    ...
}
```
In your case, `handler.selectAll()` will only concerns rows in the current page, right?


Working on it

**Comment by sgoth** [comment]
Yes, for selectAll i only use current page scope. That i implemented myself as remote api doesn't support it yet (i just iterate all items on the page and call handler.select).

Currently i just patched out the line in PageHandler as i couldn't find a hook that could preserve the data.

The selection model works perfectly fine for me then as i have real IDs for each row (independent from page). No change needed for that. Your suggestion might even break it for me as I don't have totalPageCount for my data model. I just want a list of IDs.

But I also need to maintain an extended selection model myself as i need to have the selected row data even when on another page.

All works fine for me without PageHandler clearing the selection.

**Comment by vincjo** [comment]
Thank you for your feedback, a great help.

I was able to implement the selection across pages in `1.13.1`. 
It's still fresh, maybe a few things are missing.
In your case you can keep your implementation or use the API one, I don't know which will be more practical.

At least you'll need to define an additional parameter at the instantiation level: `selectionScope`
```ts
const handler = new DataHandler(data, { rowsPerPage: 10, selectionScope: 'acrossPages' })
```

You'll find a live example by following this link:
https://vincjo.fr/datatables/remote/examples/beer-api

**Comment by sgoth** [comment]
Thank you for working on this!
When trying to switch to your solution i noticed you implemented it by storing the selection per page now ('acrossPages').
I think it would be better to store the IDs in a flat array.

As is, it leads to visualization bugs when filters or rowsPerPage changes the page an item belongs to.
For the sake of ID based selection, the actual page the item is on (or if any at all) doesn't really matter.

Repro on Beer API example:
- select all on page 1
- select first item on page 2
![Screenshot_20231115_115731](https://github.com/vincjo/datatables/assets/1681610/534b592d-5a8d-458b-986c-13ad77ff203c)

- Filter by item name of page2 item 💥:
Item is not selected visually but is in model (clearing filter shows that)
![Screenshot_20231115_115845](https://github.com/vincjo/datatables/assets/1681610/84799260-4d47-4f0a-928f-99a8f0ee447e)

- On page 1 change to 20 per page 💥:
![image](https://github.com/vincjo/datatables/assets/1681610/a318eb63-a552-4477-a031-2a755b0764d7)

**Comment by vincjo** [comment]
Indeed, thx. Wasn't the right choice. 
i'm going to change the way of checking equalities

**Comment by vincjo** [comment]
This should be better in `1.14.0`

I removed the "acrossPages" option and the default behaviour now is a sticky selection as you suggested.
https://vincjo.fr/datatables/remote/examples/beer-api

I added a global parameter that allows you to define the name of the column containing the unique identifiers for each row. 

```ts
const handler = new DataHandler( data, { rowsPerPage: 10, selectBy: 'id' })
```
This parameter is at the instantiation level because it is required to set the derived `isAllSelected()` store.

It is necessary to use unique identifiers for the comparison to work correctly. The equality between objects is verified by instance instead of reference which excludes their use in a context of server-side data processing.

**Comment by sgoth** [comment]
That works as expected! Thanks!
